website/content/docs/nia/tasks.mdx

```diff
@@ -49,7 +61,18 @@ An enabled task is executed when any change of information for services the task
 | `node_tagged_addresses` | List of explicit LAN and WAN IP addresses for the agent                                           |
 | `node_meta`             | List of user-defined metadata key/value pairs for the node                                        |
 
-Consul-Terraform-Sync automatically generates any files needed to execute the network driver for each task. See [network drivers](/docs/nia/network-drivers) for more details on the files generated for the Terraform driver.
+
+### Catalog-Services Condition
+
+Tasks with a catalog-services condition monitor and execute on service registration changes for services that satisfy the condition configuration. 'Service registration changes' specifically refers to service registration and deregistration where service registration occurs on the first service instance registration, and service deregistration occurs on the last service instance registration. Tasks with a catalog-services condition may, depending on the module, additionally monitor but not execute on service instance information.
+
+The catalog-services condition operates by monitoring the [Catalog List Services API](/api-docs/catalog#list-services) and executing the task when services are added or removed in the list of registered services. Note, the task does not execute on changes to the tags of the list of services.
+
+The module selected for a task will likely determine whether the task should be configured with a catalog-services condition. Authors for a catalog-services condition module have the option to include and use the [`catalog_services` input variable](/docs/nia/terraform-modules#catalog-services-variable). When this variable is included in the module, the user of this module will need to configure the task condition's [`source_includes_var`](/docs/nia/configuration#source_includes_var) value to be true. Catalog-Services condition modules should have the condition type and `source_includes_var` configuration value documented for users to reference when configuring a task.
```

Comment:
```suggestion
The module selected for a task will likely determine whether the task should be configured with a catalog-services condition. Module authors have the option to include and use the [`catalog_services` input variable](/docs/nia/terraform-modules#catalog-services-variable), in addition to the required `services` variable. When this variable is included in the module, the user of this module will need to configure the task condition's [`source_includes_var`](/docs/nia/configuration#source_includes_var) value to be true. Catalog-Services condition modules should have the condition type and `source_includes_var` configuration value documented for users to reference when configuring a task.
```

"catalog-services condition module" was long to read and I don't know how catchy that would be to describe a module that uses the catalog-services variable.

I'm slightly concerned about ambiguity for readers about the requirements of these variables, so add a little bit to reference that var.services is still required. This is a suggestion, and you're welcome to rephrase or omit if you think it'll add confusion mentioning another variable in this section.

Reply:
Sounds good! I appreciate your wording and will add it. I think it's good to clarify the required services variable, which isn't mentioned here, and since this is the 'catalog-services condition' section, I don't think it's necessary to elaborate on the module.